elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/filter/expression/HavingFilterExpression.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.filter.expression;
+
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpressionVisitor;
+
+/**
+ * A sub-type of {@link FilterExpression} that is evaluated in-memory on top of returned data from persistent storage.
+ * <p>
+ * This is a {@link java.util.function functional interface} whose functional method is
+ * {@link #accept(FilterExpressionVisitor)}
+ */
+@FunctionalInterface
+public interface HavingFilterExpression extends FilterExpression {
```

Comment:
I'm not sure these are needed.  We already have fields in the executeQuery interface that differentiates the where and having bits. I would leave those arguments as just FilterExpressions